airbyte-cdk/python/unit_tests/sources/test_abstract_source.py

```diff
@@ -98,9 +98,13 @@ def primary_key(self) -> Optional[Union[str, List[str], List[List[str]]]]:
 class MockStreamWithState(MockStream):
     cursor_field = "cursor"
 
+    def __init__(self, inputs_and_mocked_outputs: List[Tuple[Mapping[str, Any], Iterable[Mapping[str, Any]]]], name: str, state=None):
+        super().__init__(inputs_and_mocked_outputs, name)
+        self._state = state
+
     @property
     def state(self):
-        return {}
+        return self._state
```

Comment:
set a state to verify the emitted state is the same as the input state when no slices are read